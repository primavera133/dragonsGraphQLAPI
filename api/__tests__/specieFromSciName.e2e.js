const { createTestClient } = require('apollo-server-testing')
const gql = require('graphql-tag')
const nock = require('nock')

const { createServer, toPromise } = require('./__utils')

const GET_SPECIE_FROM_SCIENTIFIC_NAME_QUERY = gql`
  query specieFromScientificName($scientific_name: String!) {
    specieFromScientificName(scientific_name: $scientific_name) {
      items_id
      scientific_name
      author_citation
      local_names
      behaviour
      description
      distribution
      habitat
      flight_period
      size {
        length
        wingspan
      }
      similar_species
      red_list {
        habitats_directive
        red_list_EU27
        red_list_europe
        red_list_mediterranean
        EU27_endemic
        red_list_europe_endemic
        trend_europe
      }
      images {
        cloud_name
        all {
          public_id
          caption
          license
          lic_url
          by
          url
        }
      }
      sources
      links {
        label
        link
      }
    }
  }
`

describe('Server - e2e: SpecieFromScientificName', () => {
  let service, graphql

  beforeEach(async () => {
    const testServer = await createServer({
      path: '/graphql'
    })
    service = testServer.service
    graphql = testServer.executeOperation
  })

  afterEach(() => service.close())

  it('gets a specie from scientific_name', async () => {
    const res = await toPromise(
      graphql({
        query: GET_SPECIE_FROM_SCIENTIFIC_NAME_QUERY,
        variables: { scientific_name: 'Aeshna affinis' }
      })
    )
    expect(res).toMatchSnapshot()
  })
})
