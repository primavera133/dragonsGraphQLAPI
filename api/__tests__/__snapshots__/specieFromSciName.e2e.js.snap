// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server - e2e: SpecieFromScientificName gets a specie from scientific_name 1`] = `
Object {
  "data": Object {
    "specieFromScientificName": Object {
      "behaviour": "This species is a powerfull flyer and often migrates from the southernmost areas of Europe far up to the north.

The male usually incessantly patrols its territory but it is not uncommon that it alights on 
tall grass and bushes.

It is the only species of Aeshna in which the male stays attached to the female after mating, laying eggs.",
      "description": "A small Aeshna, similar in size to A. Mixta, which it resembles in many ways.

The males have bright blue eyes and facial suture barely marked. 
Thorax is yellow-brown with blue hues in mature males.

In contrast with other species of the family Aeshnidae like A. cyanea and A. viridis, 
antehumeral stripes are short and narrow, 
abdomen is characterised by large light-blue spots and by a distinctive black pattern
on the second abdominal segment, S2.
The pterostigma is relatively long.
The Costa, wing front, has a light colour.

The females can be very similar to males. 
A more lightly yellow jizz compared to other females can be noted.
They can be distinguished from those of A. mixta by the relatively short abdominal appendages.
The rather uni-coloured sides of the abdomen of females make them look a lot like females of A. viridis.  ",
      "distribution": "A largely western paleatctic speciea, confined to north-west africa, Europe and south-western Asia.s
In Europe it is rarely abundant and present mainly in the south-east.
It is relatively uncommon in many areas, including lerge parts of the Iberian peninsula, and to a lesser extent, Italy.
It is common in much of south-east Europe and locally abundant i Ukraine.",
      "flight_period": "May to August",
      "habitat": "It occurs in environments with standing or slow running water with abundant vegetaion along the banks. 
It also seems to favour waters on the verge of drying out. 
It is rarely observed in quarries, gravel pits and above 700 m.",
      "items_id": "f2acd97bbac584258f1a680bf206653b",
      "local_names": Array [
        "Southern migrant hawker",
        "Blue-eyed hawker",
      ],
      "red_list": Object {
        "EU27_endemic": "No",
        "habitats_directive": "No",
        "red_list_EU27": "Least Concern",
        "red_list_europe": "Least Concern",
        "red_list_europe_endemic": "No",
        "red_list_mediterranean": "Least Concern",
        "trend_europe": "Increasing",
      },
      "scientific_name": "Aeshna affinis",
      "similar_species": Array [
        "Aeshna mixta",
        "Aeshna viridis (females)",
      ],
      "size": Object {
        "length": "57-66 mm",
        "wingspan": "74-84 mm",
      },
    },
  },
}
`;
